src/runtime/pmix_progress_threads.c

```diff
@@ -19,7 +19,12 @@
 #ifdef HAVE_UNISTD_H
 #    include <unistd.h>
 #endif
+
 #include <pthread.h>
+#if (defined(__FreeBSD__) || defined(__OpenBSD__)) && defined(HAVE_PTHREAD_SETAFFINITY_NP)
```

Comment:
I'm a little leery of these OS-specific checks in the code - we generally do them in the Autoconf configure code. For example, in this case we would just check for the existence of the `pthread_np.h` header. I'm not sure why we need to also check for the `HAVE_PTHREAD_SETAFFINITY_NP` definition - is there some reason why we would need that?

Is there an additional required library? It seems a little odd that a function would be defined in a different header, but the actual function would reside in the same `pthread` library.

Reply:
> About the checking for `HAVE_PTHREAD_SETAFFINITY_NP` that is used to guard the use of the `_np` extensions further below. My assumption would be that we use the same guard when including the header

No, that shouldn't be required. If the header exists, then we include it. The code is simply protected against the case where it doesn't since that variable is defined in the `pthread_np.h` header.